From db6499e41e9fe311a2ae4141fc223817d0eb76ae Mon Sep 17 00:00:00 2001
From: Krystian Plata <k.plata@metrological.com>
Date: Mon, 30 Aug 2021 13:39:46 +0000
Subject: [PATCH] BCMCZ-359 Add configuration of discard thresholds

This commit introduces handling for two new env vars:
  - GST_BRCM_AUDIO_DISCARD_THRESHOLD,
  - GST_BRCM_VIDEO_DISCARD_THRESHOLD,
which override the default settings for audio / video thresholds.

Change-Id: Ie431bba412cf40309d056ea9c5bb8d450be455b2
---
 reference/audiodecode/src/gst_brcm_audio_decoder.c | 7 +++++++
 reference/videodecode/src/gst_brcm_video_decoder.c | 5 +++++
 2 files changed, 12 insertions(+)

diff --git a/reference/audiodecode/src/gst_brcm_audio_decoder.c b/reference/audiodecode/src/gst_brcm_audio_decoder.c
index 8b2df24..ec47c4e 100755
--- a/reference/audiodecode/src/gst_brcm_audio_decoder.c
+++ b/reference/audiodecode/src/gst_brcm_audio_decoder.c
@@ -474,6 +474,13 @@ static void gst_brcm_audio_decoder_init(GstBrcmAudioDecoder *decoder) {
     settings.secondary.streamStatusAvailable.callback = pt_status_available_cb;
     settings.secondary.streamStatusAvailable.context = decoder;
 
+    const gchar* discardThresholdEnv = g_getenv("GST_BRCM_AUDIO_DISCARD_THRESHOLD");
+    if (discardThresholdEnv) {
+        settings.primary.discardThreshold = atoi(discardThresholdEnv);
+        settings.secondary.discardThreshold = settings.primary.discardThreshold;
+        settings.description.discardThreshold = settings.primary.discardThreshold;
+    }
+
     settings.primary.ptsError.callback = pts_error_cb;
     settings.primary.ptsError.context = decoder;
 
diff --git a/reference/videodecode/src/gst_brcm_video_decoder.c b/reference/videodecode/src/gst_brcm_video_decoder.c
index 10c5cec..2133eb5 100755
--- a/reference/videodecode/src/gst_brcm_video_decoder.c
+++ b/reference/videodecode/src/gst_brcm_video_decoder.c
@@ -2034,6 +2034,11 @@ static GstStateChangeReturn gst_brcm_video_decoder_change_state(GstElement *elem
 
         settings.ptsOffset =  decoder->pts_offset;
 
+        const gchar* discardThresholdEnv = g_getenv("GST_BRCM_VIDEO_DISCARD_THRESHOLD");
+        if (discardThresholdEnv) {
+            settings.discardThreshold = atoi(discardThresholdEnv);
+        }
+
         NEXUS_SimpleVideoDecoder_SetSettings(decoder->video_decoder, &settings);
         NEXUS_SimpleVideoDecoder_SetExtendedSettings(decoder->video_decoder, &ext_settings);
         decoder->start_after_link = 0;
-- 
2.7.4

